stablehlo/dialect/StablehloOps.td

```diff
@@ -1948,9 +1948,9 @@ def StableHLO_CholeskyOp : StableHLO_Op<"cholesky",
 }
 
 def StableHLO_ClampOp : StableHLO_ShapedInterfaceOp<"clamp", [Pure,
-  SameOperandsAndResultElementType, HLO_BroadcastingElementwise,
+  SameOperandsAndResultElementType /* clamp_c3 */, HLO_BroadcastingElementwise,
```

Comment:
Can you include the following Trait for C4?:
```
AllTypesMatch<["operand", "result"]> /*clamp_c4*/
```

Reply:
Good advise! The reason to not having this is to support dynamism, so that we need be careful of using `AllTypesMatch` which does not support "?" dim. A relative choice is `HLO_CompatibleOperandsAndResultType` but it assert all operands. So let's keep it as is (inference infra guard this as well).

Reply:
Makes sense, thanks.